insonmnia/hub/server.go

```diff
@@ -1007,18 +1043,32 @@ func (h *Hub) processClusterEvent(value interface{}) {
 		//We don't care now
 	case LeadershipEvent:
 		//We don't care now
-	case *map[string]*TaskInfo:
+	case map[string]*TaskInfo:
 		log.G(h.ctx).Info("synchronizing tasks from cluster")
 		h.tasksMu.Lock()
-		h.tasks = *value
-		h.tasksMu.Unlock()
+		defer h.tasksMu.Unlock()
+		h.tasks = value
+	case map[string]DeviceProperties:
+		h.devicePropertiesMu.Lock()
+		defer h.devicePropertiesMu.Unlock()
+		h.deviceProperties = value
+	case []*structs.Slot:
+		h.slotsMu.Lock()
+		defer h.slotsMu.Unlock()
+		h.slots = value
+	case ACLStorage:
+		h.acl = value
 	case error:
 		// continue reading from event channel as there could be pending events
 		go func() {
 			log.G(h.ctx).Warn("cluster failure, retrying after 10 seconds", zap.Error(value))
 			time.Sleep(time.Second * 10)
 			h.eventCh = h.cluster.Run()
```

Comment:
is it safe to change h.eventCh w/o lock?

Reply:
Reworked this part